content/kubermatic/main/installation/upgrading/upgrade-from-2.25-to-2.26/_index.en.md

```diff
@@ -125,10 +125,121 @@ nginx v1.10 brings quite a few potentially breaking changes:
+    annotations: {}
+
+  certIssuer:
+    name: letsencrypt-prod
+    kind: ClusterIssuer
+```
+{{% /tab %}}
+
+{{% tab name="new dex Chart" %}}
+```yaml
+# Tell the KKP installer to install the new dex Chart into the
+# "dex" namespace, instead of the old oauth Chart.
+useNewDexChart: true
+
+dex:
+  replicaCount: 2
+
+  envVars: []
+  volumes: []
+  volumeMounts: []
+
+  ingress:
+    enabled: true
+    className: "nginx"
+
+    annotations:
+      cert-manager.io/cluster-issuer: letsencrypt-prod
+
+    hosts:
+      # Required: host must be set; usually this is the same
+      # host as is used for the KKP Dashboard, but it can be
+      # any other name.
+      - host: "kkp.example.com"
+        paths:
+          - path: /dex
+            pathType: ImplementationSpecific
+    tls:
+      - secretName: dex-tls
+        hosts:
+          # Required: must include at least the host chosen
+          # above.
+          - "kkp.example.com"
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+Additionally, Dex's own configuration is now more clearly separated from how Dex's Kubernetes manifests are configused. The following changes are required:
+
+* In general, Dex's configuration is everything under `dex.config`.
+* `dex.config.issuer` has to be set explicitly (the old `oauth` Chart automatically set it), usually to `https://<dex host>/dex`, e.g. `https://kkp.example.com/dex`.
+* `dex.connectors` is now `dex.config.connectors`
+* `dex.expiry` is now `dex.config.expiry`
+* `dex.frontend` is now `dex.config.frontend`
+* `dex.grpc` is now `dex.config.grpc`
+* `dex.clients` is now `dex.config.staticClients`
+* `dex.staticPasswords` is now `dex.config.staticPasswords` (when using static passwords, you also have to set `dex.config.enablePasswordDB` to `true`)
+
+Finally, theming support has changed. The old `oauth` Helm chart allowed to inline certain assets, like logos, as base64-encoded blobs into the Helm values. This mechanism is not available in the new `dex` Helm chart and admins have to manually provision the desired theme. KKP's Dex chart will setup a `dex-theme-kkp` ConfigMap, which is mounted into Dex and then overlays files over the default theme that ships with Dex. To customize, create your own ConfigMap/Secret and adjust `dex.volumes`, `dex.volumeMounts` and `dex.config.frontend.theme` / `dex.config.frontend.dir` accordingly.
+
+Once you have prepared a new `values.yaml` with the updated configuration, remember to set `useNewDexChart` to `true` and then you're ready. The next time you run the KKP installer, it will install the `dex` Chart for you,but leave the `oauth` release untouched in your cluster. Note that you cannot have two Ingress objects with the same host names and paths, so if you install the new Dex in parallel to the old one, you will have to temporarily use a different hostname (e.g. `kkp.example.com/dex` for the old one and `kkp.example.com/dex2` for the new Dex installation).
```

Comment:
```suggestion
Once you have prepared a new `values.yaml` with the updated configuration, remember to set `useNewDexChart` to `true` and then you're ready. The next time you run the KKP installer, it will install the `dex` Chart for you, but leave the `oauth` release untouched in your cluster. Note that you cannot have two Ingress objects with the same host names and paths, so if you install the new Dex in parallel to the old one, you will have to temporarily use a different hostname (e.g. `kkp.example.com/dex` for the old one and `kkp.example.com/dex2` for the new Dex installation).
```